test/Tests.Infrastructure/AmazonGlacierRetryTheoryAttribute.cs

```diff
@@ -37,35 +37,23 @@ static AmazonGlacierRetryTheoryAttribute()
             }
         }
 
+        public AmazonGlacierRetryTheoryAttribute()
+        {
+        }
+
         public AmazonGlacierRetryTheoryAttribute([CallerMemberName] string memberName = "", int maxRetries = 3, int delayBetweenRetriesMs = 0, params Type[] skipOnExceptions)
             : base(maxRetries, delayBetweenRetriesMs, skipOnExceptions)
         {
-            if (RavenTestHelper.SkipIntegrationTests)
-            {
-                Skip = RavenTestHelper.SkipIntegrationMessage;
-                return;
-            }
-            
-            //if (RavenTestHelper.IsRunningOnCI)
-            //    return;
-
-            if (EnvVariableMissing)
-            {
-                Skip = $"Test is missing '{GlacierCredentialEnvironmentVariable}' environment variable.";
-                return;
-            }
+        }
 
-            if (string.IsNullOrEmpty(ParsingError) == false)
+        public override string Skip
+        {
+            get
             {
-                Skip = $"Failed to parse the Amazon Glacier settings, error: {ParsingError}";
-                return;
+                return AzureRetryTheoryAttribute.TestIsMissingCloudCredentialEnvironmentVariable(EnvVariableMissing, GlacierCredentialEnvironmentVariable, ParsingError, _glacierSettings, skipIsRunningOnCICheck: true);
```

Comment:
don't we need to set it in ctor once?

Reply:
I added this so in case we will want to pass it to `MultiTheoryAttribute` it will compile since it need default ctor, so `Activator.CreateInstance(t)` won't throw.

Reply:
we can but need to do it for parameterless ctor as well, so I decided to override `Skip` property as we do in other Attribute classes